### DISCUSSION TITLE
Create a more general subscription package

### DISCUSSION THREAD

**Discussion by ilijaNL** [enisdenjo/graphql-ws#461] [Ideas]
Hello, 
First thanks for these awesome library(ies). 

I was thinking about to create a more general package for ws subscriptions based on this protocol without any relation to graphql. Looking at the implementation of the server code the graphql code dependency can be removed which only leaves  2 required functions: `execute` and `subscribe` for handling the incoming subscriptions/requests.

The general use-case could be that additional tools like trpc or any other similar packages can re-use these packages.

Wonder what your thoughts are @enisdenjo . 

PS: I am willing to contribute/rewrite.

**Comment by enisdenjo** [comment]
It sounds like a cool idea, I've thought about a similar concept before, but never got around doing it...

I'd love to see your take on it though! 👀

**Reply by ilijaNL** [reply]
> It sounds like a cool idea, I've thought about a similar concept before, but never got around doing it...
> 
> I'd love to see your take on it though! 👀

I made a seperate branch without any graphql dependency: https://github.com/ilijaNL/graphql-ws/tree/ws-graphql-transport

It just follows the protocol defined here but can be used with any execution implementation and subscribe payload (see: https://github.com/ilijaNL/graphql-ws/blob/fc6282c75cea504a63f75d92ce02922b977e9a6d/src/common.ts#L115)

 However I am not sure what the smart follow up is on this idea. Since the code is still the same for like 85% I really don't like the idea of having 2 repos with almost identical code. Additionally I really like the amount of tests you made for the client & the server, which is a must have for these kind of libraries. I wonder what your thoughts are @enisdenjo

**Reply by ilijaNL** [reply]
In meantime I made seperate package: https://www.npmjs.com/package/@ilijanl/graphql-ws-graphqless which is based on https://github.com/ilijaNL/graphql-ws

**Reply by enisdenjo** [reply]
Nice one! Will glance over it when I get some free time. :D
